src/lib.rs

```diff
@@ -63,34 +62,36 @@ impl ToOperation for TodoOperation {
     }
 }
 
-pub fn operation_from_string(s: String) -> TodoOperation {
-    let vec: Vec<&str> = s.split('\0').collect();
-
-    TodoOperation {
-        operation: match vec[0] {
-            "add" => OperationType::Add,
-            "remove" => OperationType::Remove,
-            _ => panic!("expected valid operation type, got {}", vec[0]),
-        },
-        todo: TodoData {
-            engine: vec[1].to_string(),
-            rule_id: vec[2].to_string(),
-            file_path: vec[3].to_string(),
-            range: Range {
-                start: Position {
-                    line: i32::from_str_radix(vec[4], 10).expect("valid range.start.line"),
-                    column: i32::from_str_radix(vec[5], 10).expect("valid range.start.column"),
-                },
-                end: Position {
-                    line: i32::from_str_radix(vec[6], 10).expect("valid range.end.line"),
-                    column: i32::from_str_radix(vec[7], 10).expect("valid range.end.column"),
+impl From<&str> for TodoOperation {
+    fn from(s: &str) -> Self {
+        let vec: Vec<&str> = s.split('\0').collect();
+
+        TodoOperation {
+            operation: match vec[0] {
+                "add" => OperationType::Add,
+                "remove" => OperationType::Remove,
+                _ => panic!("expected valid operation type, got {}", vec[0]),
+            },
+            todo: TodoData {
+                engine: vec[1].to_string(),
+                rule_id: vec[2].to_string(),
+                file_path: vec[3].to_string(),
+                range: Range {
+                    start: Position {
+                        line: vec[4].parse::<i32>().expect("valid range.start.line"),
+                        column: vec[5].parse::<i32>().expect("valid range.start.column"),
+                    },
+                    end: Position {
+                        line: vec[6].parse::<i32>().expect("valid range.end.line"),
+                        column: vec[7].parse::<i32>().expect("valid range.end.column"),
```

Comment:
All of these suggestions came from Clippy!

Reply:
Hmm, have I got clippy setup wrong? I thought I am doing `cargo clippy` in CI 🤔

Reply:
Dang it. I started setting it up, but never finished. I’m currently calling `cargo check` in CI.

https://github.com/ember-template-lint/todo-operation-parser/blob/9789ed887cc92b51927c085e9addf10bc81cd611/.github/workflows/ci.yml#L27

I’ll change that to `cargo clippy` separately (since cargo clippy calls cargo check IIRC)

Reply:
It's not an error, only a suggestion. I have my rust-analyzer set up to use `clippy` for its check-on-save command, which in turn shows suggestions like this as well as lint errors.

Reply:
Ah, gotcha. I’ll dig into that. I do have rust-analyzer all setup for my dev environment so it should be possible for me to configure…